pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -202,10 +221,35 @@ func (configgen *ConfigGeneratorImpl) BuildDeltaClusters(proxy *model.Proxy, pus
 		model.XdsLogDetails{AdditionalInfo: fmt.Sprintf("cached:%v/%v", clusterCacheStats.hits, clusterCacheStats.hits+clusterCacheStats.miss)}, true
 }
 
+// getRemovedClusterName finds the name of the removed cluster given a hostname and list of cluster names.
+func getRemovedClusterName(resNames []string, target string) string {
+	for _, n := range resNames {
+		_, _, svcHost, _ := model.ParseSubsetKey(n)
```

Comment:
say we have dr for foo.com with subsets=[a,b]. We remove it.

This would also remove `outbound|80||foo.com` I think? when it shouldn't, should only remove  `outbound|80|a|foo.com`  and  `outbound|80|b|foo.com` 

On that note - do we handle the subsets list changing? That would end up with some deleted/added clusters